interfaces/cython/cantera/_onedim.pyx

```diff
@@ -342,8 +345,8 @@ cdef class Inlet1D(Boundary1D):
     domain - it must be either the leftmost or rightmost domain in a
     stack.
     """
-    def __cinit__(self, *args, **kwargs):
-        self.inlet = new CxxInlet1D()
+    def __cinit__(self, _SolutionBase phase, **kwargs):
```

Comment:
The `__cinit__` functions should retain `*args` and `**kwargs` in addition to any specific arguments. Otherwise, it is impossible to introduce derived types that take additional arguments since `__cinit__` is always called with the original argument list.